tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Package/SdkCodeTool.cs

```diff
@@ -0,0 +1,467 @@
+            tspConfigPath = tspConfigPath.Trim();
+            
+            // Validate inputs
+            logger.LogInformation($"Generating SDK at repo: {localSdkRepoPath}");
+            if (string.IsNullOrEmpty(localSdkRepoPath) || !Directory.Exists(localSdkRepoPath))
+            {
+                return CreateFailureResponse($"The directory for the local sdk repo does not provide or exist at the specified path: {localSdkRepoPath}. Prompt user to clone the matched SDK repository users want to generate SDK against.");
+            }
+
+            // Get the generate script path
+            string sdkRepoRoot = gitHelper.DiscoverRepoRoot(localSdkRepoPath);
+            if (string.IsNullOrEmpty(sdkRepoRoot))
+            {
+                return CreateFailureResponse($"Failed to discover local sdk repo with path: {localSdkRepoPath}.");
+            }
+
+            // Validate arguments for local tspconfig.yaml case
+            if (!tspConfigPath.StartsWith("https://", StringComparison.OrdinalIgnoreCase))
+            {
+                var validationResponse = ValidateLocalTspConfig(tspConfigPath, specCommitSha, specRepoFullName);
+                if (validationResponse != null)
+                {
+                    return validationResponse;
+                }
+            }
+            else
+            {
+                logger.LogInformation($"Remote 'tspconfig.yaml' URL detected: {tspConfigPath}.");
+                if (!IsValidRemoteGitHubUrlWithCommit(tspConfigPath))
+                {
+                    return CreateFailureResponse($"Invalid remote GitHub URL with commit: {tspConfigPath}. The URL must include a valid commit SHA. Example: https://github.com/Azure/azure-rest-api-specs/blob/dee71463cbde1d416c47cf544e34f7966a94ddcb/specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml");
+                }
+                // For remote tspconfig.yaml case, clear specCommitSha and specRepoFullName
+                specCommitSha = string.Empty;
+                specRepoFullName = string.Empty;
+            }
+
+            return await RunTspInit(localSdkRepoPath, tspConfigPath, specCommitSha, specRepoFullName, ct);
+        }
+
+        // Run tsp-client update command to re-generate the SDK code
+        private async Task<DefaultCommandResponse> RunTspUpdate(string tspLocationPath, CancellationToken ct)
+        {
+            if (!File.Exists(tspLocationPath))
+            {
+                return CreateFailureResponse($"The 'tsp-location.yaml' file does not exist at the specified path: {tspLocationPath}");
+            }
+
+            logger.LogInformation($"Running tsp-client update command in directory: {Path.GetDirectoryName(tspLocationPath)}");
+
+            var tspLocationDirectory = Path.GetDirectoryName(tspLocationPath);
+            var npxOptions = new NpxOptions(
+                "@azure-tools/typespec-client-generator-cli",
+                ["tsp-client", "update", "--save-inputs"],
```

Comment:
`-y` is needed to prevent `npx` from asking for confirmation in the scenario where the package is not already installed:

```suggestion
                ["-y", "tsp-client", "update", "--save-inputs"],
```

That said, maybe we should add that flag to `NpxHelper` itself since I can't think of any time we wouldn't want it.

Reply:
Added to NpxOptions